src/runtime_src/core/tools/xbutil2/SubCmdStatus.cpp

```diff
@@ -0,0 +1,605 @@
+/**
+ * Copyright (C) 2019-2020 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "SubCmdStatus.h"
+#include "tools/common/XBUtilities.h"
+namespace XBU = XBUtilities;
+
+#include "core/include/xrt.h"
+
+#include "core/common/system.h"
+#include "core/common/device.h"
+#include "core/common/error.h"
+#include "core/common/xrt_profiling.h"
+#include "core/include/xcl_axi_checker_codes.h"
+
+// 3rd Party Library - Include Files
+#include <boost/program_options.hpp>
+#include <boost/format.hpp>
+namespace po = boost::program_options;
+
+// System - Include Files
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <iomanip>
+
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+
+const int8_t maxDebugIpType = TRACE_S2MM_FULL+1;
+
+bool debugIpOpt[maxDebugIpType];
+uint64_t debugIpNum[maxDebugIpType];
+static const char* debugIpNames[maxDebugIpType] = {
+  "unknown",
+  "LAPC",
+  "ILA",
+  "AIM",
+  "TraceFunnel",
+  "TraceFifoLite",
+  "TraceFifoFull",
+  "AM",
+  "ASM",
+  "AxiStreamProtocolChecker",
+  "TS2MM",
+  "AxiDMA",
+  "TS2MMFull"
+};
+
+
+uint32_t getDebugIpData(debug_ip_layout* map, int type,
+    std::vector<uint64_t> *baseAddress, std::vector<std::string> * portNames) 
+{
+  uint32_t count = 0;
+  for(uint64_t i = 0; i < map->m_count; i++) {
+    if(map->m_debug_ip_data[i].m_type != type)
+      continue;
+
+    if (baseAddress)
+      baseAddress->push_back(map->m_debug_ip_data[i].m_base_address);
+    if(portNames) {
+      std::string portName;
+      // Fill up string with 128 characters (padded with null characters)
+      portName.assign(map->m_debug_ip_data[i].m_name, 128);
+      // Strip away any extraneous null characters
+      portName.assign(portName.c_str());
+      portNames->push_back(portName);
+    }
+    ++count;
+  }
+  return count;
+}
+
+
+std::pair<size_t, size_t> getCUNamePortName (std::vector<std::string>& aSlotNames,
+                             std::vector< std::pair<std::string, std::string> >& aCUNamePortNames) 
+{
+  //Slotnames are of the format "/cuname/portname" or "cuname/portname", split them and return in separate vector
+  //return max length of the cuname and port names
+  size_t max1 = 0, max2 = 0;
+  char sep = '/';
+  for (auto slotName: aSlotNames) {
```

Comment:
auto&.   It's not for free to copy a std::string and you don't seem to be modifying slotName